vowpalwabbit/parse_args.cc

```diff
@@ -1003,6 +1006,13 @@ void parse_reductions(vw& all)
   all.reduction_stack.push_back(bs_setup);
 
   all.l = setup_base(all);
+
+  if (all.probabilities)
+  { if (!all.vm.count("oaa") && !all.vm.count("csoaa_ldf"))
+      THROW("--probabilities can only be used with --oaa=N or --csoaa_ldf=mc");
+    if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
```

Comment:
I would prefer the definition of --probabilities in oaa.cc and csoaa.cc 
and doing the checks locally.  Modularity at a small cost in extra code 
seems worthwhile.  If you can remove the probabilities flag from the 
global datastructure and just rely on the local datastructure for each, 
that seems good to.

-John

On 10/28/2015 12:25 PM, Martin Popel wrote:
>
> In vowpalwabbit/parse_args.cc 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837#discussion_r43279766>:
>
> > @@ -1003,6 +1006,13 @@ void parse_reductions(vw& all)
> >    all.reduction_stack.push_back(bs_setup);
> >
> >    all.l = setup_base(all);
> > +
> > +  if (all.probabilities)
> > +  { if (!all.vm.count("oaa") && !all.vm.count("csoaa_ldf"))
> > +      THROW("--probabilities can only be used with --oaa=N or --csoaa_ldf=mc");
> > +    if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
>
> I would need to repeat the test both in oaa.cc and csoaa.cc. Should I 
> do it?
> I felt the test (just two lines of code) fits here where I am testing 
> |all.probabilities && (!oaa || !csoaa_ldf)| (and this test obviously 
> cannot be moved to oaa.cc/csoaa.cc).
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837/files#r43279766>.
>